bus-mapping/src/circuit_input_builder.rs

```diff
@@ -377,11 +458,53 @@ impl<'a> CircuitInputStateRef<'a> {
             .map(|s| s.stack.last().unwrap_or_else(|_| Word::zero()))
             .unwrap_or_else(Word::zero);
 
-        // Return from a call with a failure, without calling RETURN
-        if step.op != OpcodeId::RETURN
-            && step.depth != next_depth
-            && next_result == Word::zero()
-        {
+        // Return from a call with a failure
+        if step.depth != next_depth && next_result == Word::zero() {
```

Comment:
We don't support the case for when `&& next_result does not hold?` I imagine that this should never happen, but we should probably handle it JIC and panic or whatever we thing is the best.

Reply:
Good point!  I've added a check for "return from a call with success without calling RETURN", which is an unexpected case.